packages/verkle/src/node/internalNode.ts

```diff
@@ -85,35 +103,64 @@ export class InternalNode extends BaseVerkleNode<VerkleNodeType.Internal> {
       // on the next byte in both keys, a recursion into
       // the moved leaf node can occur.
       const nextByteInExistingKey = child.stem[this.depth + 1]
-      const newBranch = InternalNode.create(this.depth + 1)
+      const newBranch = InternalNode.create(this.depth + 1, this.verkleCrypto)
       newBranch.cowChild(nextByteInExistingKey)
-      this.children[childIndex] = newBranch
+      this.children[childIndex] = newBranch.commitment
       newBranch.children[nextByteInExistingKey] = child
       child.depth += 1
 
       const nextByteInInsertedKey = stem[this.depth + 1]
       if (nextByteInInsertedKey === nextByteInExistingKey) {
-        return newBranch.insertStem(stem, values, resolver)
+        return newBranch.insertStem(stem, values, resolver, verkleCrypto)
       }
 
       // Next word differs, so this was the last level.
       // Insert it directly into its final slot.
-      const leafNode = LeafNode.create(stem, values)
+      // TODO: Fix this following `trie.put` logic
+      let leafCommitment = verkleCrypto.zeroCommitment
+      let c1 = verkleCrypto.zeroCommitment
+      let c2 = verkleCrypto.zeroCommitment
+      for (const [idx, value] of values.entries()) {
+        if (bytesToBigInt(value) > BIGINT_0) {
+          leafCommitment = verkleCrypto.updateCommitment(
+            leafCommitment,
+            idx,
+            new Uint8Array(32),
+            value
+          )
+          if (idx < 128) {
+            // We multiply the commitment index by 2 here because each 32 byte value in the leaf node is represented as two 16 byte arrays
+            c1 = verkleCrypto.updateCommitment(c1, idx * 2, new Uint8Array(32), value)
+          } else {
+            c2 = verkleCrypto.updateCommitment(c2, (idx - 128) * 2, new Uint8Array(32), value)
+          }
+        }
+      }
 
+      const leafNode = LeafNode.create(
+        stem,
+        values,
+        this.depth + 1,
+        leafCommitment,
+        c1,
+        c2,
+        this.verkleCrypto
+      )
+
+      // TODO - Why is the leaf node set at depth + 2 instead of + 1)?
       leafNode.setDepth(this.depth + 2)
       newBranch.cowChild(nextByteInInsertedKey)
       newBranch.children[nextByteInInsertedKey] = leafNode
     } else if (child instanceof InternalNode) {
       this.cowChild(childIndex)
-      return child.insertStem(stem, values, resolver)
+      return child.insertStem(stem, values, resolver, verkleCrypto)
     } else {
       throw new Error('Invalid node type')
     }
   }
 
   // TODO: go-verkle also adds the bitlist to the raw format.
```

Comment:
```suggestion
```

We can do without the bitlist as it was only used in the geth serialization implementation for their own purposes, and will be supserseded by an updated, more efficient serialization technique, see: 
- https://github.com/ethereum/go-verkle/pull/433 
- https://github.com/ethereum/go-verkle/pull/432